### ISSUE TITLE
Gradient accumulate steps cause bugs

### ISSUE THREAD

**Issue by Soulcrzkc** [bmaltais/kohya_ss#735]
Currently, Gradient accumulate steps are multiplied by epochs, causing the number of epochs not to be controlled by the "epoch" parameter. Max_train_epoch can solve this problem, but the warm-up step is calculated based on epochs, which causes the training process to always be in warm-up.
 I'm not sure if should fix the calculation bug of Gradient accumulate steps or simply modify the calculation method of max_train_steps? I don't know where exactly the problem lies.
I set epoch to 30 and didn't set Max_train_epoch will be:
![AR~L7BCZF56@%6SSUQFZ9OW](https://user-images.githubusercontent.com/15704468/236665592-36d20e6b-81ad-467e-99b0-aa33e928c805.png)
if I set Max_train_epoch, it will be:
![IG(DN2$8NHK4}JUQ9SVU)$N](https://user-images.githubusercontent.com/15704468/236665682-278fc1cc-5210-4469-9ad4-f44e9fb2a612.png)
but the lr will be:
![image](https://user-images.githubusercontent.com/15704468/236665694-3e76e492-3e0d-4ea5-a7ef-e0edde345d32.png)

**Comment by bmaltais** [comment]
I think this is an issue with how kohya_ss calculate steps... You should raise the issue on his repo.

**Comment by Soulcrzkc** [comment]
> I think this is an issue with how kohya_ss calculate steps... You should raise the issue on his repo.

Someone has already submitted a PR on the other side, but there might be a minor issue with the max_train_steps here. It will cause the lr_warmup_steps wrong

**Comment by bmaltais** [comment]
How would you suggest the math logic be changed? I could probably implement it to calculate the max steps differently when gradient accumulate is used. Should the total steps be multiplied by the gradient accumulate value?

**Comment by Soulcrzkc** [comment]
I think it should be divided by the gradient_accumulation_steps.
![image|50](https://user-images.githubusercontent.com/15704468/236679701-87a30807-eabd-4680-ba53-dad0839fc474.png)

**Comment by bmaltais** [comment]
Sound good, I will add the division by the gradient accumulation steps to the various UI tabs.

**Comment by bmaltais** [comment]
I have pushed a commit to the dev branch that should take care of this. Give it a try.

**Comment by Soulcrzkc** [comment]
> I have pushed a commit to the dev branch that should take care of this. Give it a try.

Thank you!
